doc/sphinx/writing-proofs.rst

```diff
@@ -0,0 +1,28 @@
+.. _writing-proofs:
+
+==============
+Writing proofs
+==============
+
+Coq is an interactive proof assistant, which means that proofs can be
+constructed interactively through a dialog between the user and the
+assistant.  The building blocks for this dialog are tactics which the
+user will use to represent steps in the proof of a theorem.  Tactics
+apply when goals have some specific forms and build a partial proof
```

Comment:
Need more explanation here, too sketchy.

Maybe add something like:

Incomplete proofs have one or more open (unproven) sub-goals.  Each goal has a set of assumptions that apply to it.  Tactics rewrite goals and assumptions.  Internally, the incomplete proof is represented as a partial proof term, with holes for the unproven sub-goals.